erpnext/accounts/report/gross_profit/gross_profit.py

```diff
@@ -335,7 +335,7 @@ def get_buying_amount(self, row, item_code):
 		else:
 			my_sle = self.sle.get((item_code, row.warehouse))
 			if (row.update_stock or row.dn_detail) and my_sle:
-				parenttype, parent = row.parenttype, row.parent
+				parenttype, parent = row.parenttype, (row.parent_invoice if row.parenttype == "Sales Invoice" and row.parent_invoice else row.parent)
```

Comment:
@mro-admin the issues caused by reusing the `parent` key seems to be fixed in this PR https://github.com/frappe/erpnext/pull/28483. Not sure if we still need this change. If you are still getting 0 valuation rates there might be some other issue. Is this only seen for product bundle items, then maybe its the parent being set as `None` while making product bundle rows